cylc/uiserver/schema.py

```diff
@@ -248,14 +254,205 @@ class Arguments:
     result = GenericScalar()
 
 
+async def get_jobs(root, info, **args):
+    if args['live']:
+        return await get_nodes_all(root, info, **args)
+
+    _, field_ids = process_resolver_info(root, info, args)
+
+    if hasattr(args, 'id'):
+        args['ids'] = [args.get('id')]
+    if field_ids:
+        if isinstance(field_ids, str):
+            field_ids = [field_ids]
+        elif isinstance(field_ids, dict):
+            field_ids = list(field_ids)
+        args['ids'] = field_ids
+    elif field_ids == []:
+        return []
+
+    for arg in ('ids', 'exids'):
+        # live objects can be represented by a universal ID
+        args[arg] = [Tokens(n_id, relative=True) for n_id in args[arg]]
+    args['workflows'] = [
+        Tokens(w_id) for w_id in args['workflows']]
+    args['exworkflows'] = [
+        Tokens(w_id) for w_id in args['exworkflows']]
+
+    return await list_jobs(args)
+
+
+async def list_jobs(args):
+    if not args['workflows']:
+        raise Exception('At least one workflow must be provided.')
+    from cylc.flow.workflow_db_mgr import WorkflowDatabaseManager
+    from pathlib import Path
+    from cylc.flow.workflow_files import get_workflow_srv_dir
+
+    jobs = []
+    for workflow in args['workflows']:
+        db_file = Path(get_workflow_srv_dir(workflow['workflow']), 'db')
+        wdbm = WorkflowDatabaseManager(db_file.parent)
+        with wdbm.get_pri_dao() as pri_dao:
+            # TODO: support all arguments including states
+            for row in pri_dao.connect().execute('''
```

Comment:
> @MetRonnie Could you help convert this to use the public DB with the latest syntax.

Part 2:

```suggestion
        with CylcWorkflowDAO(db_file, is_public=True) as dao:
            # TODO: support all arguments including states
            for row in dao.connect().execute('''
```

(Only the cylc-flow scheduler should really need a fully fledged `WorkflowDatabaseManager`, everything else should use the `CylcWorkflowDAO` (Database Access Object))